sdk/python/tests/compiler/testdata/custom_task_params.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2020 kubeflow.org
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+from kfp import dsl
+
+custom_task_name = "some-custom-task"
+custom_task_api_version = "custom.tekton.dev/v1alpha1"
+custom_task_image = "some-image"
+custom_task_command = "cmd"
+custom_task_kind = "custom-task"
+
+
+def custom_task_args(foo: str, bar: Any, pi: float) -> list:
+    return [
+        "--foo",
+        foo,
+        "--bar",
+        bar,
+        "--pi",
+        pi,
+    ]
+
+
+custom_task_results = ["target"]
+custom_task_resource_name = "some-custom-resource"
+custom_task_resource = {
+    "params": [
+        {
+            "name": "foo",
+            "type": "string",
+        },
+        {
+            "name": "bar",
+        },
+        {
+            "name": "pi",
+            "type": "int",
+            "default": 3
+        }
+    ]
+}
+
+from kfp_tekton.tekton import TEKTON_CUSTOM_TASK_IMAGES
+TEKTON_CUSTOM_TASK_IMAGES = TEKTON_CUSTOM_TASK_IMAGES.append(custom_task_image)
+
+
+def custom_task(resource_label: str, foo: str, bar: Any, pi: float) -> dsl.ContainerOp:
+    task = dsl.ContainerOp(
```

Comment:
FYI, when KFP switching to the V2 (IR) compiler next year, this dsl.ContainerOp will no longer work. So we are slowly migrating the ContainerOp DSL out of our samples and tests. https://github.com/kubeflow/kfp-tekton/issues/670

Since we are still in the middle of this migration, we don't have to change it for this PR. But we should be aware of it for future changes.